adapters/beachfront/beachfront.go

```diff
@@ -0,0 +1,541 @@
+		beachfrontReqs.Banner, err = getBannerRequest(req)
+		if err != nil {
+			return beachfrontReqs, err
+		}
+	} else {
+		beachfrontReqs.Video, err = getVideoRequest(req)
+		if err != nil {
+			return beachfrontReqs, err
+		}
+	}
+
+	return beachfrontReqs, err
+}
+
+func getBannerRequest(req *openrtb.BidRequest) (BeachfrontBannerRequest, error) {
+	var beachfrontReq BeachfrontBannerRequest
+	dec := json.NewDecoder(strings.NewReader(beachfrontBannerRequestTemplate))
+	for {
+		if err := dec.Decode(&beachfrontReq); err == io.EOF {
+			break
+		} else if err != nil {
+			// possible banner error 1 - decoding error
+			return beachfrontReq, err
+		}
+	}
+
+	// step through the prebid request "imp" and inject into the beachfront request.
+	for i, imp := range req.Imp {
+		if imp.Video != nil {
+			return beachfrontReq, nil
+		} else if imp.Banner != nil {
+			if i > 0 {
+				beachfrontReq.Slots = append(beachfrontReq.Slots, BeachfrontSlot{})
+				beachfrontReq.Slots[i].Sizes = append(beachfrontReq.Slots[i].Sizes, BeachfrontSize{})
+			}
+			for j := range imp.Banner.Format {
+				if j > 0 {
+					// multi-banner.json test
+					beachfrontReq.Slots[i].Sizes = append(beachfrontReq.Slots[i].Sizes, BeachfrontSize{})
+				}
+				beachfrontReq.Slots[i].Sizes[j].H = imp.Banner.Format[j].H
+				beachfrontReq.Slots[i].Sizes[j].W = imp.Banner.Format[j].W
+			}
+
+			beachfrontReq.Slots[i].Bidfloor = imp.BidFloor
+
+			var bidderExt adapters.ExtImpBidder
+			if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+				// possible banner error 2
+				return beachfrontReq, err
+			}
+
+			var beachfrontExt openrtb_ext.ExtImpBeachfront
+			if err := json.Unmarshal(bidderExt.Bidder, &beachfrontExt); err != nil {
+				// possible banner error 3 - supplemental/unmarshal-error-banner.json
+				return beachfrontReq, err
+			}
+
+			if req.Device != nil {
+				beachfrontReq.Ip = req.Device.IP
+				beachfrontReq.DeviceModel = req.Device.Model
+				beachfrontReq.DeviceOs = req.Device.OS
+				beachfrontReq.Dnt = req.Device.DNT
+				if req.Device.UA != "" {
+					beachfrontReq.Ua = req.Device.UA
+				}
+			}
+
+			beachfrontReq.Slots[i].Slot = req.Imp[i].ID
+
+			beachfrontReq.Slots[i].Id = beachfrontExt.AppId
+		}
+	}
+
+	if req.User != nil {
+		beachfrontReq.User = req.User.BuyerUID
+	}
+
+	if(req.App != nil) {
+		beachfrontReq.Domain = req.App.Domain
+		beachfrontReq.Page = req.App.ID
+	} else {
+		beachfrontReq.Domain = strings.Split(strings.Split(req.Site.Page, "//")[1], "/")[0]
+		beachfrontReq.Page = req.Site.Page
+	}
+
+	return beachfrontReq, nil
+}
+
+/*
+Prepare the request that has been received from Prebid.js, translating it to the beachfront format
+*/
+func getVideoRequest(req *openrtb.BidRequest) (BeachfrontVideoRequest, error) {
+	var beachfrontReq BeachfrontVideoRequest
+	var videoImps int8 = 0
+	dec := json.NewDecoder(strings.NewReader(beachfrontVideoRequestTemplate))
+
+	for {
+		if err := dec.Decode(&beachfrontReq); err == io.EOF {
```

Comment:
See my comment above about decoding a default banner template in `getBannerRequest()`. It all should apply to this video request template.